[KevinEyo1] iP

KevinEyo1 opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ **_SUPER_ FAST** to use
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

KevinEyo1 committed:
Level-1 done

KevinEyo1 committed:
Level-2 done

KevinEyo1 committed:
Level-3 done

KevinEyo1 committed:
Level-4 done

KevinEyo1 committed:
A-TextUiTesting done

KevinEyo1 committed:
Level-5 done

KevinEyo1 committed:
Level-6 done

KevinEyo1 committed:
Add change

KevinEyo1 committed:
Do Level-7

KevinEyo1 committed:
Do Level-8

KevinEyo1 committed:
Redo

KevinEyo1 committed:
Merge branch 'branch-Level-7'

KevinEyo1 committed:
Merge branch 'branch-Level-8'

KevinEyo1 committed:
Resolve conflicts

KevinEyo1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

KevinEyo1 committed:
Add A-MoreOOP

KevinEyo1 committed:
Add A-JUnit

KevinEyo1 committed:
Add A-JavaDoc

KevinEyo1 committed:
Add A-CodingStandard

KevinEyo1 committed:
Add level-9

KevinEyo1 committed:
Merge branch 'branch-A-JavaDoc'

KevinEyo1 committed:
Resolve conflicts

KevinEyo1 committed:
Resolve conflicts

linustws commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,85 @@
+        if (cmd.equals("mark")) {
+            if (len == 1) {
+                throw new DukeException("Unable to mark.");
+            }
+            return new MarkCommand(details);
+        } else if (cmd.equals("unmark")) {
+            if (len == 1) {
+                throw new DukeException("Unable to unmark.");
+            }
+            return new UnmarkCommand(details);
+        } else if (cmd.equals("todo")) {
+            if (len == 1) {
+                throw new DukeException("Description of todo cannot be empty.");
+            }
+            return new AddTodoCommand(details);
+        } else if (cmd.equals("deadline")) {
+            String[] temp = details.split(" /by ");
+            if (len == 1 || temp.length != 2) {
+                throw new DukeException("Description of deadline cannot be incomplete.");
+            }
+            return new AddDeadlineCommand(temp[0], temp[1]);
+        } else if (cmd.equals("event")) {
+            String[] temp = details.split(" /from ");
+            if (len == 1 || temp.length != 2) {
+                throw new DukeException("Description of event cannot be incomplete.");
+            }
+            String[] temp2 = temp[1].split(" /to ");
+            if (temp2.length != 2) {
+                throw new DukeException("Description of event cannot be incomplete.");
+            }
+            return new AddEventCommand(temp[0], temp2[0], temp2[1]);
+        } else if (cmd.equals("delete")) {
+            if (len == 1) {
+                throw new DukeException("Unable to delete.");
+            }
+            return new DeleteCommand(details);
+        } else if (cmd.equals("find")) {
+            return new FindCommand(details);
+        } else {
+            throw new DukeException("I do not quite understand that.");
```
Perhaps can consider using switch case instead of if-else? I heard that it's faster, especially when the list gets longer.

KevinEyo1 replied:
You are right!

> Perhaps can consider using switch case instead of if-else? I heard that it's faster, especially when the list gets longer.

linustws commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,35 @@
+    /**
+     * Executes the command given by the user.
+     *
+     * @param tasks   to be modified
+     * @param ui      to display changes
+     * @param storage to interact with as necessary
+     * @throws DukeException
```
Should the parameter description start with caps and have a fullstop at the end? Also, should they be indented at the same level or be just right after the parameter name? Perhaps can add why the DukeException might be thrown at the back of @throws, according to the Java coding standard. I noticed this same issue in several other places too.

KevinEyo1 replied:
The coding standard seems to state as such.

Good point pointing out about the @throws.

linustws commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,34 @@
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        ui.showMatchingTasks(tasks.filterTasks(this.findString));
```
I like how you separated the ui and the tasklist such that there's a command for the ui to print each respective action that the tasklist executed, instead of letting the tasklist print it. This embodies the concept of abstraction. 👍

KevinEyo1 replied:
Thanks!

> I like how you separated the ui and the tasklist such that there's a command for the ui to print each respective action that the tasklist executed, instead of letting the tasklist print it. This embodies the concept of abstraction. 👍

linustws reviewed (commented):
LGTM, almost. Just a few nits to fix. 😃

igezt commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,159 @@
+        System.out.println(DIVIDER);
+    }
+
+}
```
Good work on abstracting out the many methods for your Ui!

KevinEyo1 replied:
Thanks!

igezt commented on src/test/java/duke/data/TaskListTest.java:
```diff
@@ -0,0 +1,29 @@
+
+public class TaskListTest {
+    @Test
+    public void testAddTask() {
```
I think you can add more details to the test method using the following convention:
whatIsBeingTested_descriptionOfTestInputs_expectedOutcome

KevinEyo1 replied:
Yes, I will take note of this.

igezt commented on src/test/java/duke/data/TaskListTest.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    @Test
+    public void testDeleteTask() {
```
Same as above! Adding more details will help you remember what the testing method is testing

KevinEyo1 replied:
Could you could provide an example?

igezt replied:
I meant using whatIsBeingTested_descriptionOfTestInputs_expectedOutcome for the naming

igezt commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,93 @@
+                String type = input[0];
+                String mark = input[1];
+                Task x = null;
+                if (type.equals("T")) {
```
Can use a switch-case statement here!

KevinEyo1 replied:
Yes, you are right!

igezt reviewed (commented):
Overall good! Use switch-case instead of if-else. And name your test-cases a little better!

KevinEyo1 committed:
Add CheckStyle

KevinEyo1 committed:
Merge branch 'branch-A-CheckStyle'

KevinEyo1 committed:
Add Level-10

KevinEyo1 committed:
Merge branch 'branch-Level-10'

KevinEyo1 committed:
Add things

KevinEyo1 committed:
Add A-Assertions

KevinEyo1 committed:
Add A-Assertions

Not using assertions for error checking.

Leaves code error prone.

Let's add assertions at key points of code to allow checking of core invariants.

Assertions not at key points do not provide much value.

KevinEyo1 committed:
Add A-CodeQuality

Code quality has some room for improvement.

Can be hard to maintain and understand with low code quality.

Let's improve code quality where possible.

It might not be good to force code quality rules anywhere because context matters.

KevinEyo1 committed:
Merge pull request #1 from KevinEyo1/branch-A-Assertions

Branch a assertions

KevinEyo1 committed:
Merge pull request #2 from KevinEyo1/branch-A-CodeQuality

Add A-CodeQuality

KevinEyo1 committed:
Add BCD-Extension

KevinEyo1 committed:
Merge branch 'branch-BCD-Extension'

KevinEyo1 committed:
Add A-Personality

KevinEyo1 committed:
Merge branch 'branch-A-Personality'

KevinEyo1 committed:
Update README.md

KevinEyo1 committed:
Add docs

KevinEyo1 committed:
Merge branch 'master' of https://github.com/KevinEyo1/ip

KevinEyo1 committed:
Update docs

KevinEyo1 committed:
Update README.md

KevinEyo1 committed:
Update README.md
